src/main/groovy/wooga/gradle/wdk/unity/WdkPluginExtension.groovy

```diff
@@ -17,32 +17,84 @@
+    ObjectFactory getObjects() {
+        throw new NotSupportedException("")
+    }
 
-    void setAssetsDir(File reportsDir)
+    private final DirectoryProperty pluginsDir = objects.directoryProperty()
+    DirectoryProperty getPluginsDir(){
+        pluginsDir
+    }
+    void setPluginsDir(Provider<Directory> value) {
+        pluginsDir.set(value)
+    }
+    void setPluginsDir(File value) {
+        pluginsDir.set(value)
+    }
 
-    void setAssetsDir(Object reportsDir)
+    private final DirectoryProperty assetsDir = objects.directoryProperty()
+    DirectoryProperty getAssetsDir() {
+        assetsDir
+    }
+    void setAssetsDir(File value) {
+        assetsDir.set(value)
+    }
+    void setAssetsDir(Provider<Directory> value) {
+        assetsDir.set(value)
+    }
 
-    File getIOSResourcePluginDir()
+    private final DirectoryProperty iosResourcePluginDir = objects.directoryProperty()
+    DirectoryProperty getIosResourcePluginDir() {
+        iosResourcePluginDir
+    }
 
-    File getAndroidResourcePluginDir()
+    private final DirectoryProperty androidResourcePluginDir = objects.directoryProperty()
+    DirectoryProperty getAndroidResourcePluginDir() {
+        androidResourcePluginDir
+    }
 
-    File getWebGLResourcePluginDir()
+    private final DirectoryProperty webGLResourcePluginDir = objects.directoryProperty()
+    DirectoryProperty getWebGLResourcePluginDir() {
+        webGLResourcePluginDir
+    }
 
-    File getPaketUnity3dInstallDir()
+    private final DirectoryProperty paketUnity3dInstallDir = objects.directoryProperty()
+    DirectoryProperty getPaketUnity3dInstallDir() {
+        paketUnity3dInstallDir
+    }
 
-    List<String> getEditorDependenciesToMoveDuringTestBuild()
+    private final List<String> editorDependeciesToMoveDuringTestBuild = new ArrayList<String>()
+    List<String> getEditorDependenciesToMoveDuringTestBuild() {
+        editorDependeciesToMoveDuringTestBuild
+    }
 
-    WdkPluginExtension editorDependenciesToMoveDuringTestBuild(String... var1)
+    WdkPluginExtension editorDependenciesToMoveDuringTestBuild(String... dependencies) {
+        if (dependencies == null) {
+            throw new IllegalArgumentException("dependencies == null!")
+        }
+        editorDependeciesToMoveDuringTestBuild.addAll(Arrays.asList(dependencies))
+        return this
+    }
 
-    WdkPluginExtension editorDependenciesToMoveDuringTestBuild(Iterable<String> var1)
+    WdkPluginExtension editorDependenciesToMoveDuringTestBuild(Iterable<String> dependencies) {
+        GUtil.addToCollection(editorDependeciesToMoveDuringTestBuild, dependencies)
+        return this
+    }
 
-    void setEditorDependenciesToMoveDuringTestBuild(Iterable<String> dependencies)
-}
\ No newline at end of file
+    void setEditorDependenciesToMoveDuringTestBuild(Iterable<String> dependencies) {
+        editorDependeciesToMoveDuringTestBuild.clear()
+        editorDependeciesToMoveDuringTestBuild.addAll(dependencies)
+    }
```

Comment:
We need to talk about this if we still need this functionality

Reply:
Aye.